test/parallel/test-stream2-httpclient-response-end.js

```diff
@@ -11,7 +11,9 @@ const server = http.createServer(function(req, res) {
     let data = '';
     res.on('readable', common.mustCall(function() {
```

Comment:
@MylesBorins do you have a way to reliably reproduce that? I'm a bit puzzled. 

I'm pretty sure this is covered in streams, and it might be an http problem. See:

https://github.com/nodejs/node/blob/c68ebe8436796ff73563342ae41386cfb31f2821/lib/_http_common.js#L125-L144 

I think the problem is due to https://github.com/nodejs/node/blob/11a2ca29babcb35132e7d93244b69c544d52dfe4/lib/_http_server.js#L408-L415, which is not calling `.destroy()` or even just `.push(null)` (both will cause `'readable'` to be emitted). I will assemble a PR.